pkg/skaffold/runner/generate_pipeline.go

```diff
@@ -0,0 +1,388 @@
+					{
+						Name: "source",
+						Type: tekton.PipelineResourceTypeGit,
+					},
+				},
+			},
+			Steps: []corev1.Container{
+				{
+					Name:       "run-deploy",
+					Image:      "gcr.io/k8s-skaffold/skaffold:v0.34.0",
+					WorkingDir: "/workspace/source",
+					Command:    []string{"skaffold"},
+					Args: []string{
+						"deploy",
+						"--filename", "skaffold.yaml",
+						"--profile", "oncluster",
+						"--build-artifacts", "build.out",
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+func generatePipeline(tasks []*tekton.Task) (*tekton.Pipeline, error) {
+	if len(tasks) == 0 {
+		return nil, errors.New("no tasks to add to pipeline")
+	}
+
+	// Create pipeline to tie together all tasks
+	pipeline := &tekton.Pipeline{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pipeline",
+			APIVersion: "tekton.dev/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "skaffold-pipeline",
+		},
+		Spec: tekton.PipelineSpec{
+			Resources: []tekton.PipelineDeclaredResource{
+				{
+					Name: "source-repo",
+					Type: tekton.PipelineResourceTypeGit,
+				},
+			},
+			Tasks: []tekton.PipelineTask{},
+		},
+	}
+	// Create tasks in pipeline spec for all corresponding tasks
+	for i, task := range tasks {
+		pipelineTask := tekton.PipelineTask{
+			Name: fmt.Sprintf("%s-task", task.Name),
+			TaskRef: tekton.TaskRef{
+				Name: task.Name,
+			},
+			RunAfter: []string{},
+			Resources: &tekton.PipelineTaskResources{
+				Inputs: []tekton.PipelineTaskInputResource{
+					{
+						Name:     "source",
+						Resource: "source-repo",
+					},
+				},
+			},
+		}
+		if i > 0 {
+			pipelineTask.RunAfter = []string{pipeline.Spec.Tasks[i-1].Name}
+		}
+		pipeline.Spec.Tasks = append(pipeline.Spec.Tasks, pipelineTask)
+	}
+
+	return pipeline, nil
+}
+
+func createSkaffoldProfile(config *latest.SkaffoldConfig) error {
+	fmt.Println("Checking for oncluster skaffold profile...")
+	profileExists := false
+	for _, profile := range config.Profiles {
+		if profile.Name == "oncluster" {
+			profileExists = true
+			break
+		}
+	}
+
+	// Check for existing oncluster profile, if none exists then prompt to create one
+	if profileExists {
+		fmt.Println("profile \"oncluster\" found!")
+		return nil
+	}
+
+confirmLoop:
+	for {
+		fmt.Print("No profile \"oncluster\" found. Create one? [y/n]: ")
+		response, err := inputReader.ReadString('\n')
+		if err != nil {
+			return errors.Wrap(err, "reading user confirmation")
+		}
+
+		response = strings.ToLower(strings.TrimSpace(response))
+		switch response {
+		case "y", "yes":
+			break confirmLoop
+		case "n", "no":
+			return nil
+		}
+	}
+
+	fmt.Println("Creating skaffold profile \"oncluster\"...")
+	profile, err := generateProfile(config)
+	if err != nil {
+		return errors.Wrap(err, "generating profile \"oncluster\"")
+	}
+
+	bProfile, err := yamlv2.Marshal([]*latest.Profile{profile})
+	if err != nil {
+		return errors.Wrap(err, "marshaling new profile")
+	}
+
+	fileContents, err := ioutil.ReadFile("skaffold.yaml")
```

Comment:
the user may have passed in a different path to their skaffold.yaml, which you can find in the runcontext!

Reply:
i think its `runCtx.Opts.Config`